ivadomed/loader/utils.py

```diff
@@ -501,3 +503,81 @@ def reorient_image(arr, slice_axis, nib_ref, nib_ref_canonical):
     trans_orient = nib.orientations.ornt_transform(ras_orientation, ref_orientation)
     # apply transformation
     return nib.orientations.apply_orientation(arr_ras, trans_orient)
+
+def create_bids_dataframe(loader_params, derivatives):
+    """Create a dataframe containing all BIDS image files in a bids_path and their metadata.
+
+    Args:
+        loader_params (dict): Loader parameters, see :doc:`configuration_file` for more details.
+        derivatives (bool): If True, derivatives are indexed.
+
+    Returns:
+        df (pd.DataFrame): Dataframe containing all BIDS image files indexed and their metadata.
+    """
+
+    # Get bids_path, bids_config, target_suffix and extensions from loader parameters
+    bids_path = loader_params['bids_path']
+    bids_config = loader_params['bids_config']
+    target_suffix = loader_params['target_suffix']
+    extensions = loader_params['extensions']
+
+    # Suppress a Future Warning from pybids about leading dot included in 'extension' from version 0.14.0
+    # The config_bids.json file used matches the future behavior
+    # TODO: when reaching version 0.14.0, remove the following line
+    pybids.config.set_option('extension_initial_dot', True)
+
+    # Initialize BIDSLayoutIndexer and BIDSLayout
+    # validate=True by default for both indexer and layout, BIDS-validator is not skipped
+    # Force index for subfolders starting with string "sub", and samples tsv and json files (for microscopy)
+    force_index = [d for d in os.listdir(bids_path) if (os.path.isdir(os.path.join(bids_path, d))) and (d.startswith("sub"))]
```

Comment:
I understand why the indexing of 'samples.tsv' and 'samples.json' are forced, but why do we have to force the indexing of folders starting by "sub" won't they already be indexed by pybids?

Reply:
This is because I kept the BIDS-validator "on", so files that are not BIDS-compliant are ignored. For microscopy, that means that sub folders at the root of the dataset are ignored because png and ometiff files are not compliant yet. So forcing "sub" is overkill for anat, but necessary for microscopy. When microscopy will be included in pybids and BIDS-validator, we will be able to remove the force_index.
This is not full proof because if non-compliant files are in the sub folders, they will be indexed to. I'll check if I can find a better condition.

Reply:
I updated the condition in 12a5b50. The indexing is now forced for sub-folders only if they include microscopy files based on extensions.